path_login.go

```diff
@@ -273,8 +274,35 @@ func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, r
 			principalIDs[to.String(vmss.Identity.PrincipalID)] = struct{}{}
 		}
 		// if not, look for user-assigned identities
-		for _, userIdentity := range vmss.Identity.UserAssignedIdentities {
-			principalIDs[to.String(userIdentity.PrincipalID)] = struct{}{}
+		for userIdentityID, userIdentity := range vmss.Identity.UserAssignedIdentities {
+			// Principal ID is not nil for VMSS uniform orchestration mode
+			if userIdentity.PrincipalID != nil {
+				principalIDs[to.String(userIdentity.PrincipalID)] = struct{}{}
+				continue
+			}
+
+			elements := strings.Split(userIdentityID, "/")
+			if len(elements) < 9 {
+				return fmt.Errorf("unable to parse the userIdentityID: %s", userIdentityID)
```

Comment:
Suggest to not use the variable name in the error for users:

```suggestion
				return fmt.Errorf("unable to parse the user-assigned identity resource ID: %s", userIdentityID)
```

Reply:
Done